kubectl-fdb/cmd/remove_instances.go

```diff
@@ -0,0 +1,205 @@
+		},
+		Example: `
+# Remove instances for a cluster in the current namespace
+kubectl fdb remove instances -c cluster instance-1 -i instance-2
+
+# Remove instances for a cluster in the namespace default
+kubectl fdb -n default remove instances -c cluster instance-1 instance-2
+
+# Remove instances for a cluster with the instance-id
+kubectl fdb -n default remove instances --use-instance-id -c cluster storage-1 storage-2
+`,
+	}
+
+	cmd.Flags().StringP("fdb-cluster", "c", "", "remove instance(s) from the provided cluster.")
+	cmd.Flags().BoolP("exclusion", "e", true, "define if the instances should be remove with exclusion.")
+	cmd.Flags().Bool("use-instance-id", false, "if set the operator will use the instance ID to remove it rather than the Pod name.")
+	cmd.Flags().Bool("shrink", false, "define if the removed instances should not be replaced.")
+	err := cmd.MarkFlagRequired("fdb-cluster")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	o.configFlags.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func getInstanceIDsFromPod(kubeClient client.Client, clusterName string, podNames []string, namespace string) ([]string, error) {
+	instances := make([]string, 0, len(podNames))
+	// Build a map to filter faster
+	podNameMap := map[string]bool{}
+	for _, instance := range podNames {
+		podNameMap[instance] = true
+	}
+
+	pods, err := getPodsForCluster(kubeClient, clusterName, namespace)
+	if err != nil {
+		return instances, err
+	}
+
+	for _, pod := range pods.Items {
+		if _, ok := podNameMap[pod.Name]; !ok {
+			continue
+		}
+
+		instances = append(instances, pod.Labels[controllers.FDBInstanceIDLabel])
+	}
+
+	return instances, nil
+}
+
+// removeInstances adds instances to the instancesToRemove field
+func removeInstances(kubeClient client.Client, clusterName string, instances []string, namespace string, withExclusion bool, withShrink bool, force bool) error {
+	if len(instances) == 0 {
+		return nil
+	}
+
+	shrinkMap := make(map[string]int)
+
+	if withShrink {
+		var pods corev1.PodList
+		err := kubeClient.List(ctx.Background(), &pods,
+			client.InNamespace(namespace),
+			client.MatchingLabels(map[string]string{
+				controllers.FDBClusterLabel: clusterName,
+			}))
+
+		if err != nil {
+			return err
+		}
+
+		for _, pod := range pods.Items {
+			class := controllers.GetProcessClassFromMeta(pod.ObjectMeta)
+			shrinkMap[class]++
+		}
+	}
+
+	var cluster fdbtypes.FoundationDBCluster
+	err := kubeClient.Get(ctx.Background(), client.ObjectKey{
+		Namespace: namespace,
+		Name:      clusterName,
+	}, &cluster)
+
+	if err != nil {
+		return err
+	}
+
+	for class, amount := range shrinkMap {
+		cluster.Spec.ProcessCounts.DecreaseCount(class, amount)
+	}
+
+	if !force {
+		confirmed := confirmAction(fmt.Sprintf("Remove %v from cluster %s/%s with exclude: %t and shrink: %t", instances, namespace, clusterName, withExclusion, withShrink))
+		if !confirmed {
+			print("Abort")
+			return nil
```

Comment:
I'm going to assume that returning nil here results in the cli exiting 0, which is probably undesirable. Much better IMO to put that Abort into a custom error and return that, then get a non-zero exit code to support anyone using automation.